packages/core/src/sdk/deliveryPromise/useDeliveryPromise.ts

```diff
@@ -373,6 +378,73 @@ export function useDeliveryPromise({
     []
   )
 
+  function getBadgesLabel(value: string) {
+    if (value === 'delivery') {
+      return (
+        deliveryPromiseSettings?.deliveryPromiseBadges?.delivery ??
+        'Available for delivery'
+      )
+    }
+    if (value === 'pickup-in-point') {
+      return (
+        deliveryPromiseSettings?.deliveryPromiseBadges?.pickupInPoint ??
+        'Available for pickup'
+      )
+    }
+    return value
+  }
+
+  function getDeliveryPromiseBadges() {
+    const badges: Array<{ label: string; availability: boolean }> = []
+    const availableTypeNames = deliveryPromiseBadges?.map(
+      (badge) => badge.typeName
+    )
+
+    const hasDelivery = availableTypeNames?.includes('delivery')
+    const hasPickupPoint = availableTypeNames?.includes('pickup-in-point')
+
+    // Only add unavailable badges if at least one delivery method is available
```

Comment:
suggestion: if deliveryPromiseBadges is empty or doesn't exist, we could add an early return in this function with an empty array []